### PR TITLE
feature/promotions-carousel

### LINKED ISSUE #153: feature/slideshow-for-products
## User Story: (1 point)
As a user, I want to be able to view promoted products on a slideshow when in main page.

## Brief Description
Main page will have a slideshow where promoted products will be showed one after each other in a carousel style.

## Feature Requirements
-  Image of products displayed in slideshow
-  Clickable images to direct user to product page
- Arrows on both left and right sides of the slideshow allow scrolling between slides

# Acceptance Test Requirements
- [ ] All promoted products are displayed in the slideshow
- [ ] Clicking the right arrow scrolls to the right
- [ ] Clicking the left arrow scrolls to the left
- [ ] Clicking a product image routes you to the selected product page

# Risk
- Promoted products gets deleted by seller.

### PR DESCRIPTION
**Pull request by jungsoolee1**
## Brief Description
Adding carousel to show promoted items on the home page.

## Linked Issues
- closes #153 

## Changes
- added carousel to view promoted products on home page

## Acceptance Criteria
- [x] All specifications met in feature requirements

### CODE REVIEW

**Comment by jungsoolee1** [comment]
Caroussel is complete and ready. Simply awaiting /findPromoted get request to properly get all data of promoted products. When resolved, carrousel can be merged.

**Comment by jungsoolee1** [comment]
Pull request is not ready for review!

**Comment by aspyridakos** [comment]
Item images shouldn't change aspect ratio when scaling down. 

@emma-amar the navbar still doing this weird overlap thing for login and sign up, probably a padding issue. As an alternative, we could have the login and sign up buttons be a dropdown from the profile icon on small screens to resolve this problem.

image

**Comment by Adamo-O** [comment]
> Item images shouldn't change aspect ratio when scaling down.
> 
> @emma-amar the navbar still doing this weird overlap thing for login and sign up, probably a padding issue. As an alternative, we could have the login and sign up buttons be a dropdown from the profile icon on small screens to resolve this problem.
> 
> image

Yeah, and the menu items should turn into icons a bit earlier (smaller breakpoint).
To be fair though, the product details page doesn't fully work at that breakpoint (Mobile S and Mobile M in Chrome DevTools) either. If anything, we either go through the whole website together to ensure everything is fully responsive, or omit the smallest breakpoints from the scope of the project.

**Comment by jungsoolee1** [comment]
Aspect ratio of pictures in slideshow have been modified to remain the same. Aspect ratio for images in **product grid** also seemed to change so I applied the "max-height" to their css for them remain with the same aspect ratio.

**Comment by Adamo-O** [comment]
image


> Aspect ratio of pictures in slideshow have been modified to remain the same. Aspect ratio for images in **product grid** also seemed to change so I applied the "max-height" to their css for them remain with the same aspect ratio.

image

Looks awesome! Nice product grid fix too!

**Comment by jungsoolee1** [comment]
> > Aspect ratio of pictures in slideshow have been modified to remain the same. Aspect ratio for images in **product grid** also seemed to change so I applied the "max-height" to their css for them remain with the same aspect ratio.
> 
> image
> 
> Looks awesome! Nice product grid fix too!

Should I let emma handle the navbar and just proceed to merge this?

**Comment by Adamo-O** [comment]
Yes, that can be handled in a different PR if anything.

**Comment by jungsoolee1** [comment]
Alright!

File: online-shopping-website/frontend/src/Components/Browse/Products/Slideshow.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <div className="each-slide">
+        <div>
+          <div className="CenteringProductDetails">
+            <div className="ProductDetailsShow">
+              <img className="ProductImageShow" src={props.product.picture} alt={props.product.name} />
+              <div>
+                <h3 className="ProductInfoSlide">{props.product.name}</h3>
+                <p className="ProductInfoSlide">Brand: {props.product.brand.name}</p>
+                <p className="ProductInfoSlide">Sold by: {props.product.seller.sellerName}</p>
+              </div>
```
Comment (Adamo-O) [inline online-shopping-website/frontend/src/Components/Browse/Products/Slideshow.js:23]:
```suggestion
              <div className="ProductInfoSlide">
                <h3>{props.product.name}</h3>
                <p>Brand: {props.product.brand.name}</p>
                <p>Sold by: {props.product.seller.sellerName}</p>
              </div>
```
I would suggest this, just for best practices, saving some repeated code, and a small speed increase 😎

**Review by Adamo-O — APPROVED** [review]
All's looking good! One small suggestion for best practices but super nice and clean code. Nice!

**Comment by jungsoolee1** [comment]
> All's looking good! One small suggestion for best practices but super nice and clean code. Nice!

Good suggestion!

**Comment by Adamo-O** [comment]
Thanks! 
I made one tiny change to ensure product grid always has consistent height. Right now, the height of the grid item can change as the image sizes down, but this makes sure they're always the same no matter the size of the image inside. Pretty nice!

**Comment by aspyridakos** [comment]
Don't merge yet. Found some little code cleanup fixes to push.

**Review by aspyridakos — APPROVED** [review]
Code is clean and good to go!

**Comment by jungsoolee1** [comment]
I'm a little upset that you changed the text from "Special!" to "Promotions" but I'll allow it. 😤

**Comment by jungsoolee1** [comment]
Approved by two reviewers. Will merge now.

**Pull request merged**